packages/3dom/README.md

```diff
@@ -0,0 +1,66 @@
+# 3DOM 
+
+3DOM is a library that makes it easy to run custom, possibly-untrusted scripts
+that manipulate a scene graph. 3DOM invokes scripts in a worker, and 3DOM
+scripts act on a normalized scene graph API designed in the image of the glTF
+file format.
+
+3DOM is designed to be implemented for multiple 3D libraries. To support 
+this goal, it uses a facade pattern that can be adapted for different backing
+APIs. Currently, only Three.js is has an implementation, but we hope to include
+more facades in the future.
+
+Although 3DOM was designed to support `<mnodel-viewer>`, you can use 3DOM with
+any Three.js scene.
+
+## Implementation example
+
+There are two important constructs to consider when implementing 3DOM:
+
+ - The graft: a facade that wraps over your Three.js scene
+ - The execution context: makes it easy to evaluate script in a scene graph worker
+
+The following example applies 3DOM to a Three.js scene graph and invokes script
+to operate on the scene graph from a worker:
+
+```javascript
+import {ThreeDOMExecutionContext} from '@google/3dom/lib/context.js';
+import {ModelGraft} from '@google/3dom/lib/facade/three-js/model-graft.js';
+
+import {GLTFLoader} from 'three/examples/jsm/loaders/GLTFLoader.js';
+
+const gltfLoader = new GLTFLoader();
+const modelUrl = './Astronaut.glb';
+
+gltfLoader.load(modelUrl, (gltf) => {
+  const executionContext =
+      new ThreeDOMExecutionContext(['material-properties']);
+
+  const graft = new ModelGraft(modelUrl, gltf);
```

Comment:
The `global.model.modelURI` can be measured by the script to know the URL that the model was loaded from. The host context is responsible for setting it appropriately, so if it wanted to it could trick the worker I guess.

Reply:
Okay, but how is it used by the script? Why does it need it?

Reply:
It is the sole identifying signal (for now) that distinguishes one specific model from another. To offer a hypothetical, if a single script were designed to be applied to multiple models, it might use the URL to map to a material name to pick out from the scene graph.

Reply:
I see, that makes sense. Is there any reason it needs to be a URL? Why not just `name`?

Reply:
There is no such thing as a model name in our domain apart from its URL. At some point, there may be optional metadata that suggests a name, but for now we are left wanting.

Reply:
Oh, I get it now; I was thinking this was user-exposed (like they could choose the name), but now I see it's not. Sounds good.